bootstrap/kubeadm/internal/ignition/clc/clc.go

```diff
@@ -0,0 +1,375 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package clc generates bootstrap data in Ignition format using Container Linux Config Transpiler.
+//
+// CLC configuration defined in this package will run kubeadm command by creating a /etc/kubeadm.sh script
+// file containing both pre and post kubeadm commands as well as the kubeadm command itself.
+//
+// /etc/kubeadm.sh script will be executed using kubeadm.service systemd unit, which will only happen
+// if /etc/kubeadm.yml file exists, which ensures the script will run only once, as by the end of the
+// script, /etc/kubeadm.yml is moved to /tmp filesystem, so it gets automatically cleaned up after a
+// reboot. This is to align the implementation with cloud-init, which places kubeadm configuration in
+// /tmp directory directly, which is not possible with Ignition.
+//
+// /etc/kubeadm.yml file contains generated kubeadm configuration and can be customized using pre kubeadm
+// commands if needed, as a replacement for Jinja templates supported by cloud-init, for example
+// using 'envsubst' or 'sed'.
+//
+// To override the behavior of kubeadm.service unit, one should create a override drop-in for this it
+// using AdditionalConfig field. Data from this field takes precedence and will be merged with
+// configuration generated by the bootstrap provider, overriding already defined fields following the
+// merge strategy described in https://coreos.github.io/ignition/operator-notes/#config-merging.
+package clc
```

Comment:
Is this tied to ContainerLinux or would it work in any system supporting ignition?

Reply:
It may work, but I don't think it's something we can guarantee.

Reply:
At this point I think Container Linux Configs should be viewed as specific to Flatcar Container Linux.  The Container Linux Config Transpiler only supports Ignition spec 2.x, which is dead upstream and only used in Flatcar.  Also, CLCs don't have a mechanism for specifying which distro sugar to enable, so they're hard to generalize to distros with different functionality (such as different implementations of system services).

Reply:
@enxebre any more thoughts before I mark it as resolved?

Reply:
no more feedback on my side. Thanks!